### PR TITLE
2890 show warning on missing description

### PR DESCRIPTION
**Pull request by Lakoja**
Feature step to show a warning icon on missing image descriptions for 
https://github.com/tuskyapp/Tusky/issues/2890

This is feature-wise complete but regarding visuals it is more of a request for comments:

- How to display that warning dialog when tapping the warning icon? For the buttons at the bottom of the compose view a "BottomSheetBehavior" is used.
- Is the warning icon well placed on the top?
- What color to use? It should be a warning-like color. Currently "textColorHighlight" is used which is the "brightest" I could detect.
- The icon itself is copied from the content warning button. How are those icons normally created (take from where)?

### CODE REVIEW

**Comment by connyduck** [comment]
Works & is definitely better than before.

- It needs a different icon though, the one you used is the same used for "Content warning", that is too confusing.
- I'd use tusky_red as the color
- Snackbar is better than Toastbecause it stays local within the Activity.

**Comment by Lakoja** [comment]
Icon changed (new one "missing comment").

There is now only one problem remaining: I cannot position it sensibly (higher than the bottom) and it is shown **below** the toolbar at the bottom.

File: app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt [hunk 1]
```diff
@@ -498,6 +498,11 @@ class ComposeActivity :
         binding.composeScheduleView.setListener(this)
         binding.atButton.setOnClickListener { atButtonClicked() }
         binding.hashButton.setOnClickListener { hashButtonClicked() }
+        binding.descriptionMissingWarningButton.setOnClickListener {
+            Snackbar.make(binding.composeMainScrollView, R.string.hint_media_description_missing, Snackbar.LENGTH_LONG)
+                .setAction(R.string.action_retry) {  }
```
Comment (Lakoja) [inline app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt:503]: It is not shown without an action??

**Comment by connyduck** [comment]
Check the other snackbars in ComposeActivity. You need to set some additional things so they are shown correctly.

**Comment by Lakoja** [comment]
The problem with the snackbar being under the linear layout (toolbar) at the bottom is, that that has an "elevation" (12) that is higher than that of a snackbar (6).

It does work to add another CoordinatorLayout as wrapper around the main scroll part so that the snackbar is shown above the toolbar. However that feels like a hack/workaround. Also it would be the only location the app where it is done this way.

The second solution would be to use different elevation (i. e. 5 or none) where currently 12 is used. That elevation is rather old (2018) so I'm not really sure what it is or was for.

As a third alternative I reverted to using a Toast. This does not have this elevation problem and is also used for other explanation messages. For example error_no_custom_emojis.

**Comment by Lakoja** [comment]
Ah found it now: displayTransientError() which explicitely sets an elevation...

Ok, I used that one, refactored it a bit and used it also for the emoji error.

I can also adapt the remaining two locations in ComposeActivity which still use Toast.

**Review by connyduck — APPROVED** [review]

**Pull request merged**